const {|class|} = require('../classes/{|class|}');

exports.get = async (req , res) => {

	try {

		{|class|}Target = await {|class|}.findManyAndPopulate(req.query , req._populate);

		res.success({|class|}Target);

	} catch (err){
		res.preconditionFailed();
	}

};

exports.getPagination = async (req , res) => {

	try {

		{|class|}Target = await {|class|}.findManyAndPopulate(req.query , req._populate , req._page , req._size );
		res.success({|class|}Target);

	} catch (err){
		res.preconditionFailed();
	}

};

exports.getSpecific = async (req , res) => {

	try {

		let {|class|}Target = await {|class|}.find( {_id: req._objectId } , req._populate );
		{|class|}Target = await {|class|}.wrap( {|class|}Target );
		res.success({|class|}Target);

	} catch (err){
		res.preconditionFailed();
	}

};

exports.create = async (req , res) => {

	try{

		const saved = await {|class|}.create(req.body);
		res.success(saved);

	} catch (err){
		res.repeat();
	}

};

exports.update = async (req , res) => {

	try {

		let target = await {|class|}.find({ _id : req._objectId });
		let updatedObj = await {|class|}.update(target , req.body);
		updatedObj = await {|class|}.wrap( updatedObj );
		res.success(updatedObj);

	} catch (err){
		res.preconditionFailed();
	}

};

exports.delete = async (req , res) => {

	try {

		let target = await {|class|}.find({ _id : req._objectId });
		let deletedObj = await {|class|}.deleteObj(target);
		deletedObj = await {|class|}.wrap( deletedObj );
		res.success(deletedObj);

	} catch (err){
		res.preconditionFailed();
	}

};

{|validate?|}
